libs/dyn/convert/normalize.go

```diff
@@ -77,6 +98,45 @@ func normalizeStruct(typ reflect.Type, src dyn.Value) (dyn.Value, diag.Diagnosti
 			out[k] = v
 		}
 
+		// Return the normalized value if missing fields are not included.
+		if !n.includeMissingFields {
+			return dyn.NewValue(out, src.Location()), diags
+		}
+
+		// Populate missing fields with their zero values.
+		for k, index := range info.Fields {
+			if _, ok := out[k]; ok {
+				continue
+			}
+
+			// Optionally dereference pointers to get the underlying field type.
+			ftyp := typ.FieldByIndex(index).Type
+			for ftyp.Kind() == reflect.Pointer {
+				ftyp = ftyp.Elem()
+			}
+
+			var v dyn.Value
+			switch ftyp.Kind() {
```

Comment:
We also have Time and Nil kinds, shall we handle them as well?

Reply:
This is the `reflect.Kind`, not the `dyn.Kind`.

It's the Go type of the field in the struct. But you raise a good point. We must never panic here under nominal operation. I'll slightly change this to simply ignore fields that we don't know how to handle.